xonsh/prompt/vc.py

```diff
@@ -52,23 +52,33 @@ def get_git_branch():
         branch = None
     return branch
 
+def _get_hg_root(q):
+    _curpwd = os.getcwd()
+    while True:
+        if any([b.name == '.hg' for b in os.scandir(_curpwd)]):
+            q.put(_curpwd)
+            break
+        else:
+            _oldpwd = _curpwd
+            _curpwd = os.path.split(_curpwd)[0]
+            if _oldpwd == _curpwd:
+                return False
```

Comment:
i think i touched this code too at some point, my prompt was getting too slow when i was deep in the dir tree cause of all the recursive searches up to root for .hg